backend/app/Actions/Visits/CreateVisitAction.php

```diff
@@ -0,0 +1,181 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Actions\Visits;
+
+use App\Entities\GeoPosition;
+use App\Entities\Page;
+use App\Entities\Session;
+use App\Entities\System;
+use App\Entities\Visit;
+use App\Repositories\Contracts\GeoPositionRepository;
+use App\Repositories\Contracts\PageRepository;
+use App\Repositories\Contracts\SessionRepository;
+use App\Repositories\Contracts\SystemRepository;
+use App\Repositories\Contracts\VisitorRepository;
+use Carbon\Carbon;
+use Illuminate\Support\Str;
+use Jenssegers\Agent\Agent;
+use Tymon\JWTAuth\Facades\JWTAuth;
+
+final class CreateVisitAction
+{
+    private $visitorRepository;
+    private $sessionRepository;
+    private $pageRepository;
+    private $systemRepository;
+    private $geoPositionRepository;
+
+    public function __construct(
+        VisitorRepository $visitorRepository,
+        SessionRepository $sessionRepository,
+        PageRepository $pageRepository,
+        SystemRepository $systemRepository,
+        GeoPositionRepository $geoPositionRepository
+    ) {
+        $this->visitorRepository = $visitorRepository;
+        $this->sessionRepository = $sessionRepository;
+        $this->pageRepository = $pageRepository;
+        $this->systemRepository = $systemRepository;
+        $this->geoPositionRepository = $geoPositionRepository;
+    }
+
+    public function execute(CreateVisitRequest $request): CreateVisitResponse
+    {
+        $token = Str::after($request->token(), 'Bearer ');
+        JWTAuth::setToken($token);
+        $visitorId = JWTAuth::getPayload()->get('visitor_id');
+
+        $visitor = $this->visitorRepository->getById($visitorId);
+
+        $page = $this->getOrCreatePage(
+            $visitor->website_id,
+            $request->pageTitle(),
+            $request->page()
+        );
+
+        $agent = new Agent();
+        $agent->setUserAgent($request->userAgent());
+        $system = $this->getOrCreateSystem(
+            $agent->platform(),
+            $request->device(),
+            $agent->browser(),
+            $request->resolutionHeight(),
+            $request->resolutionWidth()
+        );
+
+        $geoPosition = $this->getOrCreateGeoPosition($request->ip());
+
+        $session = $this->getOrCreateSession(
+            $visitor->id,
+            $page->id,
+            $request->language(),
+            $system->id
+        );
+
+        $visit = Visit::create([
```

Comment:
entity cannot set itself. It is problem of eloquent and active record pattern at all. It should be saved by corresponding repository